src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs

```diff
@@ -56,6 +61,12 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
                 builder.UseMiddleware<PlaceholderSpecializationMiddleware>();
             }
 
+            // Enable response compression only after PlaceholderSpecializationMiddleware, as it requires customer opt-in feature flag value.
+            builder.UseWhen(_ => responseCompressionOptions.EnableResponseCompression, config =>
```

Comment:
Any reason you are using `OnChange` hooks and not just accessing `.CurrentValue.EnableResponseCompression` in here?

Reply:
Was trying to use same pattern used for other options above. I pushed an iteration in which I switched to using CurrentValue instead of onchange and local variable (only updated the new code to use this pattern)